ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -1045,6 +1032,17 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
                         this.appChanged.next(app);
                     }
                 }
+            },
+            {
+                iconClassName: classNames('fa fa-redo', {'status-icon--spin': !!refreshing}),
+                title: <ActionMenuItem actionLabel='Invalidate Cache' />,
+                action: () => {
+                    if (!refreshing) {
```

Comment:
Should we disable the button if there's an ongoing refresh? Otherwise looks like the click gets silently ignored.